beacon-chain/core/blocks/block_operations_test.go

```diff
@@ -153,3 +156,327 @@ func TestProcessProposerSlashings_AppliesCorrectStatus(t *testing.T) {
+				Data:                           att2,
+				AggregateSignaturePoc_0Indices: []uint32{5, 6},
+				AggregateSignaturePoc_1Indices: []uint32{7, 8},
+			},
+		},
+	}
+	registry := []*pb.ValidatorRecord{}
+	currentSlot := uint64(0)
+
+	want := "expected intersection of vote indices to be non-empty"
+	if _, err := ProcessCasperSlashings(
+		registry,
+		slashings,
+		currentSlot,
+	); !strings.Contains(err.Error(), want) {
+		t.Errorf("Expected %s, received %v", want, err)
+	}
+}
+
+func TestProcessCasperSlashings_AppliesCorrectStatus(t *testing.T) {
+	// We test the case when data is correct and verify the validator
+	// registry has been updated.
+	registry := []*pb.ValidatorRecord{
+		{
+			Status:                 pb.ValidatorRecord_ACTIVE,
+			LatestStatusChangeSlot: 0,
+		},
+		{
+			Status:                 pb.ValidatorRecord_ACTIVE,
+			LatestStatusChangeSlot: 0,
+		},
+	}
+
+	att1 := &pb.AttestationData{
+		Slot:          5,
+		JustifiedSlot: 5,
+	}
+	att2 := &pb.AttestationData{
+		Slot:          5,
+		JustifiedSlot: 4,
+	}
+	slashings := []*pb.CasperSlashing{
+		{
+			Votes_1: &pb.SlashableVoteData{
+				Data:                           att1,
+				AggregateSignaturePoc_0Indices: []uint32{0, 1},
+				AggregateSignaturePoc_1Indices: []uint32{2, 3},
+			},
+			Votes_2: &pb.SlashableVoteData{
+				Data:                           att2,
+				AggregateSignaturePoc_0Indices: []uint32{4, 5},
+				AggregateSignaturePoc_1Indices: []uint32{6, 1},
+			},
+		},
+	}
+
+	currentSlot := uint64(5)
+	newRegistry, err := ProcessCasperSlashings(
+		registry,
+		slashings,
+		currentSlot,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Given the intersection of slashable indices is [1], only validator
+	// at index 1 should be penalized and change Status. We confirm this below.
+	if newRegistry[1].Status != pb.ValidatorRecord_EXITED_WITH_PENALTY {
+		t.Errorf(
+			`
+			Expected validator at index 1's status to change to 
+			EXITED_WITH_PENALTY, received %v instead
+			`,
+			newRegistry[1].Status,
+		)
+	}
+	if newRegistry[0].Status != pb.ValidatorRecord_ACTIVE {
+		t.Errorf(
+			`
+			Expected validator at index 0's status to remain 
+			ACTIVE, received %v instead
+			`,
+			newRegistry[1].Status,
+		)
+	}
+}
+
+func TestIntersection(t *testing.T) {
+	testCases := []struct {
+		setA []uint32
+		setB []uint32
+		out  []uint32
+	}{
+		{[]uint32{2, 3, 5}, []uint32{3}, []uint32{3}},
+		{[]uint32{2, 3, 5}, []uint32{3, 5}, []uint32{3, 5}},
+		{[]uint32{2, 3, 5}, []uint32{5, 3, 2}, []uint32{5, 3, 2}},
+		{[]uint32{2, 3, 5}, []uint32{2, 3, 5}, []uint32{2, 3, 5}},
+		{[]uint32{2, 3, 5}, []uint32{}, []uint32{}},
+		{[]uint32{}, []uint32{2, 3, 5}, []uint32{}},
+		{[]uint32{}, []uint32{}, []uint32{}},
+		{[]uint32{1}, []uint32{1}, []uint32{1}},
+	}
+	for _, tt := range testCases {
+		result := intersection(tt.setA, tt.setB)
+		if !testEq(result, tt.out) {
+			t.Errorf("got %d, want %d", result, tt.out)
+		}
+	}
+}
+
+func testEq(a, b []uint32) bool {
```

Comment:
If you got this from stackoverflow (I suspect you did because its exactly the same) then please comment where this came from so the author can be attributed proper credit.

https://stackoverflow.com/questions/15311969/checking-the-equality-of-two-slices

Reply:
This could be placed in the shared folder to check for equality of slices, I am pretty sure we would need it in other places in the future